data/plugins/projectsearch.lua

```diff
@@ -56,9 +56,9 @@ function ResultsView:begin_search(text, fn)
   core.add_thread(function()
     local i = 1
     for dir_name, file in core.get_project_files() do
-      if file.type == "file" then
-        local path = (dir_name == core.project_dir and "" or (dir_name .. PATHSEP))
-        find_all_matches_in_file(self.results, path .. file.filename, fn)
+      if file.type == "file" and (not path or (dir_name .. "/" .. file.filename):find(path, 1, true) == 1) then
```

Comment:
Would this work too?
```suggestion
      if file.type == "file" and (not path or (dir_name .. "/"):find(path, 1, true) == 1) then
```

Reply:
You are right again.

Reply:
Actually, no, checked this; dir_name is the project path, file.filename is the subsequent path. We'll need the whole thing to check.

Strange way of naming things, really, but I think it's outside scope to fundamentally change the naming convention.